src/main/scala/com/gyoho/adget/AdgetService.scala

```diff
@@ -0,0 +1,30 @@
+package com.gyoho.adget
+
+import com.gyoho.adget.utils.ScalaHttpClient
+
+object AdgetService {
+  @volatile private var ads: Map[String, Ad] = Map.empty
```

Comment:
Because we have only one writer and an assignment operation is atomic, `volatile` seems sufficient for the map value. See: https://stackoverflow.com/questions/44283378/why-is-reference-assignment-atomic-in-java